lib/MusicBrainz/Server/Edit/Medium/Edit.pm

```diff
@@ -439,7 +439,10 @@ sub accept {
         }
 
         my @final_tracklist;
-        my $existing_recordings = $self->c->model('Recording')->get_by_ids(@merged_recordings);
+        my $existing_recordings = $self->c->sql->select_single_column_array(
+            'SELECT id FROM recording WHERE id = any(?)', \@merged_recordings
```

Comment:
The fix is to bypass the cache and just directly check which ids exist in the table, since that's all it needs to know.